// @framer/steveruizok.undraw/code/UnDraw.tsx
import * as React from 'react'
import { PropertyControls, ControlType } from 'framer'
import Undraw from 'react-undraw'

const names = [
	'a-day-at-the-park',
	'about-us-page',
	'accept-terms',
	'account',
	'active-support',
	'add-files',
	'add-to-cart',
	'add-user',
	'address',
	'agreement',
	'air-support',
	'airport',
	'alert',
	'alien-science',
	'analysis',
	'analytics',
	'android',
	'app-installation',
	'appreciation',
	'art-lover',
	'artificial-intelligence',
	'astronaut',
	'at-work',
	'authentication',
	'autumn',
	'baby',
	'back-in-the-day',
	'back-to-school',
	'balloons',
	'barber',
	'basketball',
	'be-the-hero',
	'beach',
	'beer-celebration',
	'before-dawn',
	'bibliophile',
	'birthday-cake',
	'bitcoin-p2p',
	'bitcoin',
	'blank-canvas',
	'blog-post',
	'blogging',
	'book-lover',
	'booking',
	'brainstorming',
	'broadcast',
	'browser-stats',
	'building-blocks',
	'building',
	'bus-stop',
	'business-deal',
	'business-plan',
	'businessman',
	'businesswoman',
	'by-my-car',
	'calculator',
	'calendar',
	'campfire',
	'camping',
	'cautious-dog',
	'celebration',
	'charts',
	'chat',
	'chatting',
	'checklist',
	'chef',
	'children',
	'chilling',
	'choose',
	'city-driver',
	'cloud-hosting',
	'cloud-sync',
	'co-workers',
	'co-working',
	'coding',
	'collaboration',
	'collecting',
	'collection',
	'community',
	'confirmation',
	'confirmed',
	'connected',
	'connecting-teams',
	'container-ship',
	'content',
	'contrast',
	'control-panel',
	'conversation',
	'country-side',
	'couple',
	'create',
	'creation-process',
	'creativity',
	'credit-card-payment',
	'credit-card-payments',
	'credit-card',
	'crypto-flowers',
	'customer-survey',
	'dark-alley',
	'dark-analytics',
	'dashboard',
	'data-report',
	'data-trends',
	'data',
	'delivery',
	'departing',
	'depi',
	'design-community',
	'design-process',
	'design-tools',
	'designer-girl',
	'designer-life',
	'designer',
	'destination',
	'developer-activity',
	'development',
	'devices',
	'digital-nomad',
	'doctor',
	'doctors',
	'documents',
	'drone-delivery',
	'drone-race',
	'easter-egg-hunt',
	'eating-together',
	'electric-car',
	'email-campaign',
	'email-capture',
	'emails',
	'empty',
	'envelope',
	'ether',
	'ethereum',
	'events',
	'exams',
	'experts',
	'exploring',
	'fall-is-coming',
	'fans',
	'fast-loading',
	'fatherhood',
	'features-overview',
	'feeling-blue',
	'file-searching',
	'filing-system',
	'filter',
	'finance',
	'firmware',
	'fishing',
	'floating',
	'focus',
	'follow-me-drone',
	'for-sale',
	'forever',
	'forgot-password',
	'freelancer',
	'friendship',
	'gaming',
	'gardening',
	'gdpr',
	'getting-coffee',
	'gift',
	'gifts',
	'girls-just-wanna-have-fun',
	'goal',
	'golden-gate-bridge',
	'good-doggy',
	'grades',
	'graduation',
	'grand-slam',
	'group-chat',
	'group-selfie',
	'growing',
	'hang-out',
	'happy-birthday',
	'healthy-habit',
	'heartbroken',
	'hello',
	'high-five',
	'hiring',
	'home-run',
	'house-searching',
	'houses',
	'image-folder',
	'image-upload',
	'images',
	'in-love',
	'in-progress',
	'in-sync',
	'in-the-office',
	'in-the-pool',
	'inbox-cleanup',
	'influencer',
	'instant-support',
	'interaction-design',
	'internet-on-the-go',
	'interview',
	'invite',
	'jason-mask',
	'javascript-frameworks',
	'job-hunt',
	'jogging',
	'lighthouse',
	'live-collaboration',
	'loading',
	'login',
	'logistics',
	'lost',
	'love-is-in-the-air',
	'mail-sent',
	'mail',
	'mailbox',
	'maintenance',
	'make-it-rain',
	'maker-launch',
	'makeup-artist',
	'making-art',
	'map-dark',
	'map-light',
	'map',
	'marilyn',
	'may-the-force',
	'medicine',
	'meditation',
	'meeting',
	'messages',
	'messaging-fun',
	'messaging',
	'messenger',
	'mindfulness',
	'missed-chances',
	'mission-impossible',
	'mobile-apps',
	'mobile-browsers',
	'mobile-marketing',
	'mobile-payments',
	'mobile-testing',
	'mobile',
	'modern-woman',
	'moment-to-remember',
	'more-music',
	'morning-essentials',
	'motherhood',
	'movie-night',
	'moving-forward',
	'music',
	'navigation',
	'nerd',
	'new-message',
	'news',
	'newsletter',
	'night-calls',
	'no-data',
	'notebook',
	'notes',
	'off-road',
	'old-day',
	'on-the-office',
	'on-the-way',
	'onboarding',
	'online-shopping',
	'online-world',
	'online',
	'opened',
	'order-confirmed',
	'ordinary-day',
	'organize-photos',
	'organize-resume',
	'organizing-projects',
	'outer-space',
	'pair-programming',
	'passing-by',
	'pedestrian-crossing',
	'pen-tool',
	'people-search',
	'personal-trainer',
	'personalization',
	'photo-sharing',
	'photo',
	'photocopy',
	'photos',
	'pie-chart',
	'pilates',
	'plain-credit-card',
	'playful-cat',
	'podcast-audience',
	'podcast',
	'portfolio',
	'post',
	'posts',
	'powerful',
	'preferences',
	'presentation',
	'press-play',
	'printing-invoices',
	'processing',
	'product-hunt',
	'product-teardown',
	'product-tour',
	'profile-pic',
	'programming',
	'prototyping-process',
	'qa-engineers',
	'questions',
	'queue',
	'quitting-time',
	'reading-list',
	'real-time-sync',
	'recording',
	'refreshing',
	'relaxation',
	'relaxing-at-home',
	'report',
	'responsive',
	'resume-folder',
	'resume',
	'revenue',
	'ride-a-bicycle',
	'rising',
	'robotics',
	'romantic-getaway',
	'running-wild',
	'safe',
	'savings',
	'schedule',
	'science',
	'scooter',
	'scrum-board',
	'search-engines',
	'search',
	'secure-data',
	'secure-server',
	'security-on',
	'security',
	'segment',
	'segmentation',
	'select',
	'selfie-time',
	'selfie',
	'server-status',
	'server',
	'setup-analytics',
	'setup-wizard',
	'setup',
	'site-stats',
	'sleep-analysis',
	'smart-home',
	'smiley-face',
	'snowman',
	'social-dashboard',
	'social-growth',
	'social-ideas',
	'social-media',
	'social-networking',
	'social-strategy',
	'social-tree',
	'specs',
	'spreadsheets',
	'starman',
	'startled',
	'startup-life',
	'static-assets',
	'statistics',
	'status-update',
	'staying-in',
	'street-food',
	'studying',
	'successful-purchase',
	'sunny-day',
	'super-thank-you',
	'superhero',
	'surfer',
	'swipe-profiles',
	'sync',
	'tabs',
	'taken',
	'taking-notes',
	'target',
	'tasting',
	'teacher',
	'team-spirit',
	'team',
	'teddy-bear',
	'texting',
	'thoughts',
	'through-the-desert',
	'time-management',
	'to-do-list',
	'to-the-stars',
	'together',
	'track-and-field',
	'transfer-files',
	'trip',
	'tweetstorm',
	'typing',
	'unboxing',
	'upload',
	'uploading',
	'upvote',
	'usability-testing',
	'user-flow',
	'vault',
	'vehicle-sale',
	'version-control',
	'video-call',
	'videographer',
	'virtual-reality',
	'voice-control',
	'vr-chat',
	'walk-in-the-city',
	'wall-post',
	'wallet',
	'warning',
	'weather',
	'web-devices',
	'website-setup',
	'wedding',
	'welcome',
	'wind-turbine',
	'window-shopping',
	'windows',
	'winners',
	'winter-designer',
	'winter-olympics',
	'wireframing',
	'wishes',
	'witch',
	'woman',
	'women-day',
	'word-of-mouth',
	'work-chat',
	'work-time',
	'working-late',
	'working',
	'workout',
	'world',
	'yacht',
	'young-and-happy',
	'youtube-tutorial',
]

const optionTitles = names.map((u) =>
	u
		.split('-')
		.map((w) => w.charAt(0).toUpperCase() + w.slice(1))
		.join(' ')
)

type Props = {
	illustration: string
	color: string
	height: number
	width: number
}

export class UnDraw extends React.Component<Props> {
	render() {
		const { illustration, color, height } = this.props

		return (
			<div
				style={{
					height: '100%',
					display: 'flex',
					alignItems: 'center',
					justifyContent: 'center',
					overflow: 'hidden',
					textAlign: 'center',
				}}
			>
				<Undraw
					name={illustration}
					primaryColor={color}
					height={height.toString()}
				/>
			</div>
		)
	}

	static defaultProps: Props = {
		illustration: names[0],
		color: '#1199EE',
		height: 200,
		width: 300,
	}

	static propertyControls: PropertyControls<Props> = {
		illustration: {
			type: ControlType.Enum,
			title: 'Illustration',
			options: names,
			optionTitles,
		},
		color: {
			type: ControlType.Color,
			title: 'Color',
			defaultValue: '#1199EE',
		},
	}
}
